ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -75,6 +75,13 @@ public DdlCommandResult executeCreateStream(final CreateStreamCommand createStre
           withQuery,
           getKsqlTopic(createStream)
       );
+
```

Comment:
@big-andy-coates @spena 
I noticed that `CreateStreamCommand` did not have the boolean flag for ifNotExists set here , I thought if this approach is ok , I will add that too and fail the query if its not set and source exists

Reply:
The place to put the check is in the `CreateSourceFactory`.  You'll need to pass the `MetaStore` in to the factory's constructor.

See `DropSourceFactory` for an example of the pattern.